Cooper/ClientApp/src.v2/app/components/language-select/language-select.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, ViewEncapsulation, Input } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { LocalizationService } from '@services';
+
+@Component({
+  selector: 'coop-language-select',
+  templateUrl: './language-select.component.html',
+  styleUrls: ['./language-select.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class LanguageSelectComponent {
+  @Input() public languageKeys: string[] = this.localizationService.languageKeys;
+
+  @Input() public languages: any = this.localizationService.languages;
+  @Input() public currentLanguage: string = this.localizationService.getCurrentLanguage();
+  public onLangChangeEvent(lang: string): void {
+    this.localizationService.onLanguageChanged(lang);
```

Comment:
it's dummy component that's why we can't use services and any non-presentational logic
you can emit event and handle it and call service in parent layout